apsg/core.py

```diff
@@ -108,24 +108,28 @@ def __pow__(self, other):
         if np.isscalar(other):
             return pow(abs(self), other)
         else:
-            return self.cross(other)
+            return Vec3(np.cross(self, other))
```

Comment:
Here we should revert to original implementation as it will fails for LIn and Fol